upd: Data Analysis (#5514)
JetBrains/kotlin-web-site: docs/topics/data-analysis/data-analysis-visualization.md

If the Kotlin Notebook features are not available, ensure the plugin is enabled. For more information,
see [Set up an environment](kotlin-notebook-set-up-env.md).

To follow this tutorial:

1. Create a [new Kotlin Notebook](kotlin-notebook-create.md).

> 2. Import [Kandy](https://kotlin.github.io/kandy/welcome.html) and [Kotlin DataFrame](https://kotlin.github.io/dataframe/home.html):

Reviewer: ```suggestion
2. In your notebook, import [Kandy](https://kotlin.github.io/kandy/welcome.html) and [Kotlin DataFrame](https://kotlin.github.io/dataframe/home.html):
```
We should explain where they should perform this action.

   ```kotlin
   %use kandy
   %use dataframe
   ```

> > Make sure to run the code cell with the `%use dataframe` line before you run any other code cells

Reviewer: Similar to the other page, can we explain why this is important?

> that rely on the Kotlin DataFrame library.
>
{style="note"}

## Create a DataFrame

> To start, let's create a `DataFrame` with data to visualize. This `DataFrame` stores

Reviewer: ```suggestion
To start, let's create a DataFrame with data to visualize. This DataFrame stores
```
DataFrame doesn't need to have inline code formatting.

    "September", "October", "November",
    "December"
)
// The tempBerlin, tempMadrid, and tempCaracas variables store a list 
// with temperature values for each month
val tempBerlin =
    listOf(-0.5, 0.0, 4.8, 9.0, 14.3, 17.5, 19.2, 18.9, 14.5, 9.7, 4.7, 1.0)
val tempMadrid =
    listOf(6.3, 7.9, 11.2, 12.9, 16.7, 21.1, 24.7, 24.2, 20.3, 15.4, 9.9, 6.6)
val tempCaracas =
    listOf(27.5, 28.9, 29.6, 30.9, 31.7, 35.1, 33.8, 32.2, 31.3, 29.4, 28.9, 27.6)
```

> Now let’s create a new variable (`df`) and use

Reviewer: ```suggestion
Now let's create a new variable (`df`) and use
```
Please check that you're using the correct apostrophe symbol in your work 🙏  Sometimes it can be incorrectly converted when pasted from other tools.

> [`dataFrameOf()`](https://kotlin.github.io/dataframe/createdataframe.html#dataframeof)

Reviewer: ```suggestion
the [`dataFrameOf()`](https://kotlin.github.io/dataframe/createdataframe.html#dataframeof) function
```

to generate a DataFrame of three columns (Month, Temperature, and City):

```kotlin
val df = dataFrameOf(
    "Month" to months + months + months,
    "Temperature" to tempBerlin + tempMadrid + tempCaracas,
    "City" to List(12) { "Berlin" } + List(12) { "Madrid" } + List(12) { "Caracas" }
)
```
To preview the data, use the [`head()`](https://kotlin.github.io/dataframe/head.html) function:

```kotlin

> df.head(4) // Returns first four rows

Reviewer: ```suggestion
df.head(4) // Returns the first four rows
```

## Create a line chart

Let's create a line chart in Kotlin Notebook using the `df` DataFrame from the previous section.

> For that:

Reviewer: ```suggestion
Let's create a line chart in Kotlin Notebook using the `df` DataFrame from the previous section:
```
"For that" isn't necessary here and sounds a little unnatural, so I suggest removing it entirely :)

## Create a points chart

> Now, let's visualize the `df` DataFrame in a points (scatter) chart. For that:

Reviewer: ```suggestion
Now, let's visualize the `df` DataFrame in a points (scatter) chart:
```

## Create a bar chart

Finally, let's create a bar chart for each city.

> For that:

Reviewer: ```suggestion
Finally, let's create a bar chart for each city.:
```

> 1. Use `groupBy` to group the `DataFrame` by the `City` column. 

Reviewer: ```suggestion
1. Use the `groupBy()` function to group the DataFrame by the `City` column. 
```
